crates/host/src/lib.rs

```diff
@@ -88,40 +90,47 @@ impl<'a> TryFrom<&'a str> for ResourceRef<'a> {
     }
 }
 
-fn extract_server<'a>(resource_ref: &ResourceRef<'a>) -> Option<&'a str> {
-    match resource_ref {
-        ResourceRef::File(_) => None,
-        ResourceRef::Bindle(resource_ref) | ResourceRef::Oci(resource_ref) => {
-            resource_ref.split('/').next()
-        }
-    }
+fn oci_authority(s: &str) -> Option<&str> {
+    let (l, _) = s.split_once('/')?;
+    Some(l)
 }
 
 /// Fetch an actor from a reference.
 #[instrument(skip(actor_ref))]
 pub async fn fetch_actor(
     actor_ref: impl AsRef<str>,
     allow_file_load: bool,
-    all_registry_settings: &HashMap<String, RegistrySettings>,
+    registry_config: &HashMap<String, RegistryConfig>,
 ) -> anyhow::Result<Vec<u8>> {
-    let actor_ref = ResourceRef::try_from(actor_ref.as_ref())?;
-    let server = extract_server(&actor_ref).unwrap_or_default();
-    let default_registry_settings = RegistrySettings::default();
-    let registry_settings = all_registry_settings
-        .get(server)
-        .unwrap_or(&default_registry_settings);
-
-    match actor_ref {
-        ResourceRef::File(actor_ref) if allow_file_load => {
+    match ResourceRef::try_from(actor_ref.as_ref())? {
```

Comment:
$0.02: With the additional nesting, branching logic, indentation, and line breaks, I find these match blocks harder to follow after this refactor

Reply:
I agree with this, but also consider it a nit as well. Generally I prefer readability over conciseness (and it can go too far in either direction)

Reply:
@connorsmith256 I think this should not be addressed